src/rpc/handlers/LedgerIndex.cpp

```diff
@@ -0,0 +1,122 @@
+#include <ctime>
+#include <iomanip>
+#include <numeric>
+#include <sstream>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace rpc {
+
+LedgerIndexHandler::Result
+LedgerIndexHandler::process(LedgerIndexHandler::Input input, Context const& ctx) const
+{
+    auto const range = sharedPtrBackend_->fetchLedgerRange();
+    auto [minIndex, maxIndex] = *range;
+
+    auto const fillOutputByIndex = [&](std::uint32_t index) {
+        auto const ledger = sharedPtrBackend_->fetchLedgerBySequence(index, ctx.yield);
+        return Output{
+            .ledgerIndex = index,
+            .ledgerHash = ripple::strHex(ledger->hash),
+            .closeTimeIso = ripple::to_string_iso(ledger->closeTime)
+        };
+    };
+
+    if (!input.date)
+        return fillOutputByIndex(maxIndex);
+
+    std::vector<std::uint32_t> ledgerRange(maxIndex - minIndex + 1);
+    std::iota(ledgerRange.begin(), ledgerRange.end(), minIndex);
+
+    auto const convertISOTimeStrToTicks = [](std::string const& isoTimeStr) {
+        std::tm time = {};
+        std::stringstream ss(isoTimeStr);
+        ss >> std::get_time(&time, DATE_FORMAT_);
+        return std::chrono::system_clock::from_time_t(std::mktime(&time)).time_since_epoch().count();
+    };
+
+    auto const ticks = convertISOTimeStrToTicks(*input.date);
+
+    auto const earlierThan = [&](std::uint32_t ledgerIndex) {
+        auto const header = sharedPtrBackend_->fetchLedgerBySequence(ledgerIndex, ctx.yield);
+        auto const ledgetTime =
+            std::chrono::system_clock::time_point{header->closeTime.time_since_epoch() + ripple::epoch_offset};
+        return ticks < ledgetTime.time_since_epoch().count();
+    };
+
+    // If the given date is earlier than the first valid ledger, return lgrNotFound
+    if (earlierThan(minIndex))
+        return Error{Status{RippledError::rpcLGR_NOT_FOUND, "ledgerNotInRange"}};
+
+    auto const greaterEqLedgerIter =
+        std::lower_bound(ledgerRange.begin(), ledgerRange.end(), ticks, [&](std::uint32_t ledgerIndex, std::int64_t) {
+            return not earlierThan(ledgerIndex);
+        });
+
+    if (greaterEqLedgerIter != ledgerRange.end()) {
+        return fillOutputByIndex(std::max(*greaterEqLedgerIter - 1, minIndex));
+    }
+    return fillOutputByIndex(maxIndex);
+}
+
+LedgerIndexHandler::Input
+tag_invoke(boost::json::value_to_tag<LedgerIndexHandler::Input>, boost::json::value const& jv)
+{
+    auto input = LedgerIndexHandler::Input{};
+
+    if (jv.as_object().contains(JS(date)))
+        input.date = jv.at(JS(date)).as_string();
+
+    return input;
+}
+
+void
+tag_invoke(boost::json::value_from_tag, boost::json::value& jv, LedgerIndexHandler::Output const& output)
+{
+    auto object = boost::json::object{
+        {JS(ledger_index), output.ledgerIndex},
+        {JS(ledger_hash), output.ledgerHash},
+        {JS(close_time_iso), output.closeTimeIso},
+        {JS(validated), true},
+    };
+    jv = std::move(object);
```

Comment:
I think no need to go through temporary in this case, just directly assign to `jv`